src/services/formatting/smartIndenter.ts

```diff
@@ -9,18 +9,21 @@ namespace ts.formatting {
         }
 
         /**
-         * Computed indentation for a given position in source file
-         * @param position - position in file
-         * @param sourceFile - target source file
-         * @param options - set of editor options that control indentation
-         * @param assumeNewLineBeforeCloseBrace - false when getIndentation is called on the text from the real source file.
-         * true - when we need to assume that position is on the newline. This is usefult for codefixes, i.e.
+         * @param assumeNewLineBeforeCloseBrace
+         * `false` when called on text from a real source file.

```

Comment:
Being called with a synthesized AST. The code-fixes in particular do this, and invoke the API's in `textChanges.ts`.